src/check.ts

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env node
+/* eslint-disable no-process-exit */
+import * as path from 'path';
+
+import {logger} from './util/logger';
+import {parseIfCheckArgs} from './util/args';
+import {execPromise} from './util/helpers';
+import {getFileName, getYamlFiles, removeFileFromDirectory} from './util/fs';
+
+import {STRINGS} from './config';
+
+const {CHECKING} = STRINGS;
+
+const IfCheck = async () => {
+  const commandArgs = await parseIfCheckArgs();
+
+  console.log(`${CHECKING}\n`);
+
+  if (commandArgs.manifest) {
+    const manifest = commandArgs.manifest;
+
+    try {
+      await executeCommands(manifest, false);
+    } catch (error: any) {
+      const fileName = path.basename(manifest);
+      const executedFile = manifest.replace(fileName, `re-${fileName}`);
+      const manifestDirPath = path.dirname(manifest);
+
+      logFailureMessage(error);
+
+      await removeFileFromDirectory(`${manifestDirPath}/package.json`);
+      await removeFileFromDirectory(executedFile);
+    }
+  } else {
+    const directory = commandArgs.directory;
+    const files = await getYamlFiles(directory!);
+
+    for await (const file of files) {
+      const fileName = path.basename(file);
+      console.log(fileName);
+
+      try {
+        await executeCommands(file, true);
```

Comment:
let;s try to escape having so many nested try catch blocks

Reply:
> let;s try to escape having so many nested try catch blocks

This one is necessary because when we get an error from one manifest, it should not stop; it should continue checking the others.